### ISSUE TITLE
Cannot read property 'dropboxAccessJson' of undefined

### ISSUE THREAD

**Issue by Anzic23** [simatec/ioBroker.backitup#659]
**Describe the bug**  
after updating to 2.4.1 received such an error.

**Screenshots & Logfiles**  

```
backitup.0 | 2022-03-29 17:31:36.184 | info | cleaned everything up...
-- | -- | -- | --
backitup.0 | 2022-03-29 17:31:36.182 | error | Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.182 | error | TypeError: Cannot read property 'dropboxAccessJson' of undefined at Object.list (/opt/iobroker/node_modules/iobroker.backitup/lib/list/dropbox.js:15:156) at module.exports (/opt/iobroker/node_modules/iobroker.backitup/lib/list.js:76:36) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1095:17 at new Promise (<anonymous>) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1093:46 at Array.map (<anonymous>) at detectLatestBackupFile (/opt/iobroker/node_modules/iobroker.backitup/main.js:1093:31) at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.backitup/main.js:1342:13) at processImmediate (internal/timers.js:466:21)
backitup.0 | 2022-03-29 17:31:36.181 | error | unhandled promise rejection: Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.181 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
backitup.0 | 2022-03-29 17:31:36.181 | error | Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.181 | error | TypeError: Cannot read property 'dropboxAccessJson' of undefined at Object.list (/opt/iobroker/node_modules/iobroker.backitup/lib/list/dropbox.js:15:156) at module.exports (/opt/iobroker/node_modules/iobroker.backitup/lib/list.js:76:36) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1095:17 at new Promise (<anonymous>) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1093:46 at Array.map (<anonymous>) at detectLatestBackupFile (/opt/iobroker/node_modules/iobroker.backitup/main.js:1093:31) at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.backitup/main.js:1342:13) at processImmediate (internal/timers.js:466:21)
backitup.0 | 2022-03-29 17:31:36.180 | error | unhandled promise rejection: Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.180 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
backitup.0 | 2022-03-29 17:31:36.178 | error | Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.178 | error | TypeError: Cannot read property 'dropboxAccessJson' of undefined at Object.list (/opt/iobroker/node_modules/iobroker.backitup/lib/list/dropbox.js:15:156) at module.exports (/opt/iobroker/node_modules/iobroker.backitup/lib/list.js:76:36) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1095:17 at new Promise (<anonymous>) at /opt/iobroker/node_modules/iobroker.backitup/main.js:1093:46 at Array.map (<anonymous>) at detectLatestBackupFile (/opt/iobroker/node_modules/iobroker.backitup/main.js:1093:31) at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.backitup/main.js:1342:13) at processImmediate (internal/timers.js:466:21)
backitup.0 | 2022-03-29 17:31:36.176 | error | unhandled promise rejection: Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 17:31:36.176 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
backitup.0 | 2022-03-29 17:31:35.880 | info | [iobroker] backup was activated at 02:00 every 1 day(s)
```

**Versions:**  
 - Adapter version: 2.4.1
 - JS-Controller version: 3.3.22
 - Node version: v14.18.3
 - Operating system: debian

**Additional context**  
the same behavior on another host where the controller is 4.0.21 and Node version: v14.19.0

**Comment by Anzic23** [comment]
solved by changing the DROPBOX menu item from "custom application with long-lived tokens" to "Default Dropbox-API application for Backitup"

**Comment by EvilEls** [comment]
Same error.
Not using Dropbox after all. Never did.


```
host.xxx| 2022-03-29 16:42:16.325 | error | instance system.adapter.backitup.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
-- | -- | -- | --
backitup.0 | 2022-03-29 16:42:15.740 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
backitup.0 | 2022-03-29 16:42:15.739 | info | terminating
backitup.0 | 2022-03-29 16:42:15.739 | info | cleaned everything up...
backitup.0 | 2022-03-29 16:42:15.736 | error | Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 16:42:15.736 | error | TypeError: Cannot read property 'dropboxAccessJson' of undefined     at Object.list (/opt/iobroker/node_modules/iobroker.backitup/lib/list/dropbox.js:15:156)     at module.exports (/opt/iobroker/node_modules/iobroker.backitup/lib/list.js:76:36)     at /opt/iobroker/node_modules/iobroker.backitup/main.js:1095:17     at new Promise (<anonymous>)     at /opt/iobroker/node_modules/iobroker.backitup/main.js:1093:46     at Array.map (<anonymous>)     at detectLatestBackupFile (/opt/iobroker/node_modules/iobroker.backitup/main.js:1093:31)     at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.backitup/main.js:1342:13)     at processImmediate (internal/timers.js:466:21)
backitup.0 | 2022-03-29 16:42:15.735 | error | unhandled promise rejection: Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 16:42:15.734 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
backitup.0 | 2022-03-29 16:42:15.540 | info | [ccu] backup was activated at 01:37 every 7 day(s)
backitup.0 | 2022-03-29 16:42:15.496 | info | [iobroker] backup was activated at 02:07 every 1 day(s)
backitup.0 | 2022-03-29 16:42:15.432 | info | starting. Version 2.4.1 in /opt/iobroker/node_modules/iobroker.backitup, node: v14.19.1, js-controller: 4.0.21
host.xxx| 2022-03-29 16:42:14.520 | info | instance system.adapter.backitup.0 started with pid 92355
host.xxx| 2022-03-29 16:41:44.389 | info | Restart adapter system.adapter.backitup.0 because enabled
host.xxx| 2022-03-29 16:41:44.389 | error | instance system.adapter.backitup.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
backitup.0 | 2022-03-29 16:41:43.793 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
backitup.0 | 2022-03-29 16:41:43.793 | info | terminating
backitup.0 | 2022-03-29 16:41:43.792 | info | cleaned everything up...
backitup.0 | 2022-03-29 16:41:43.790 | error | Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 16:41:43.789 | error | TypeError:  Cannot read property 'dropboxAccessJson' of undefined    at Object.list   (/opt/iobroker/node_modules/iobroker.backitup/lib/list/dropbox.js:15:156)     at module.exports  (/opt/iobroker/node_modules/iobroker.backitup/lib/list.js:76:36)    at  /opt/iobroker/node_modules/iobroker.backitup/main.js:1095:17    at new  Promise (<anonymous>)    at  /opt/iobroker/node_modules/iobroker.backitup/main.js:1093:46    at  Array.map (<anonymous>)    at detectLatestBackupFile  (/opt/iobroker/node_modules/iobroker.backitup/main.js:1093:31)    at  Immediate.<anonymous>  (/opt/iobroker/node_modules/iobroker.backitup/main.js:1342:13)    at  processImmediate (internal/timers.js:466:21)
backitup.0 | 2022-03-29 16:41:43.788 | error | unhandled promise rejection: Cannot read property 'dropboxAccessJson' of undefined
backitup.0 | 2022-03-29 16:41:43.788 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
backitup.0 | 2022-03-29 16:41:43.583 | info | [ccu] backup was activated at 01:37 every 7 day(s)
backitup.0 | 2022-03-29 16:41:43.541 | info | [iobroker] backup was activated at 02:07 every 1 day(s)
backitup.0 | 2022-03-29 16:41:43.478 | info | starting. Version 2.4.1 in /opt/iobroker/node_modules/iobroker.backitup, node: v14.19.1, js-controller: 4.0.21
```

**Comment by simatec** [comment]
Fix in Version 2.4.2 on Github and NPM
